ts/src/pro/coinbase.ts

```diff
@@ -0,0 +1,560 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+
+import coinbaseRest from '../coinbase.js';
+import { ArgumentsRequired } from '../base/errors.js';
+import { ArrayCacheBySymbolById } from '../base/ws/Cache.js';
+import { sha256 } from '../static_dependencies/noble-hashes/sha256.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class coinbase extends coinbaseRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchOHLCV': false,
+                'watchOrderBook': true,
+                'watchTicker': true,
+                'watchTickers': true,
+                'watchTrades': true,
+                'watchBalance': false,
+                'watchOrders': true,
+                'watchMyTrades': false,
+            },
+            'urls': {
+                'api': {
+                    'ws': 'wss://advanced-trade-ws.coinbase.com',
+                },
+            },
+            'options': {
+                'tradesLimit': 1000,
+                'ordersLimit': 1000,
+                'myTradesLimit': 1000,
+                'sides': {
+                    'bid': 'bids',
+                    'offer': 'asks',
+                },
+            },
+        });
+    }
+
+    async subscribe (name, symbol = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description subscribes to a websocket channel
+         * @see https://docs.cloud.coinbase.com/advanced-trade-api/docs/ws-overview#subscribe
+         * @param {string} name the name of the channel
```

Comment:
@samgermain we need to follow the "updated" syntax for js docs, of for instance optional parameters should be surrounded by square brackets, example:
```
* @param {int} [limit]
```